### ISSUE TITLE
Incorrect `clippy::useless_attribute` for `ambiguous_glob_reexports`

### ISSUE THREAD

**Issue by SamRodri** [rust-lang/rust-clippy#10878]
### Summary

The new stable (rust 1.70) `ambiguous_glob_reexports` is valid in use items.

### Lint Name

useless_attribute

### Reproducer

I tried this code:

```rust
mod foo1 {
    pub struct Foo;
}

mod foo2 {
    pub struct Foo;
}

// #[allow(clippy::useless_attribute)]
#[allow(ambiguous_glob_reexports)]
pub use foo1::*;
pub use foo2::*;
```

I saw this happen:

Deny error in `#[allow(ambiguous_glob_reexports)]`.

I expected to see this happen:

No error

### Version

```text
rustc 1.70.0 (90c541806 2023-05-31)
binary: rustc
commit-hash: 90c541806f23a127002de5b4038be731ba1458ca
commit-date: 2023-05-31
host: x86_64-pc-windows-msvc
release: 1.70.0
LLVM version: 16.0.2
```
